### PR TITLE
Add nodelocal-dns-cache addon

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Adds the nodelocal-dns-cache addon which greatly improves resiliency of the DNS service on busy clusters

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
The resiliency of in-cluster DNS was greatly improved by adding the nodelocal-dns-cache addon, which runs a DNS cache on each node, avoiding the need to use NAT for DNS queries
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
/hold

to update image tag

File: addons/nodelocal-dns-cache/dns.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      priorityClassName: system-node-critical
+      serviceAccountName: node-local-dns
+      hostNetwork: true
+      dnsPolicy: Default  # Don't use cluster DNS.
+      tolerations:
+      - key: "CriticalAddonsOnly"
+        operator: "Exists"
+      containers:
+      - name: node-cache
+        image: {{ Registry "k8s.gcr.io" }}/k8s-dns-node-cache:1.15.2
```
Comment (mrIncompetent) [inline addons/nodelocal-dns-cache/dns.yaml]: Use `{{ Registry "gcr.io" }}/google_containers/k8s-dns-node-cache:1.15.2` instead.

`k8s.gcr.io` is just some kind of symbolic link and might not work with custom/offline registries as it omits the org.

Comment (alvaroaleman) [reply addons/nodelocal-dns-cache/dns.yaml]: Updated, PTAL

**Comment by alvaroaleman** [comment]
@mrIncompetent Bumped the addon tag, PTAL

/hold cancel

**Pull request merged**